platform/security/certificate/security-crl-generator/src/main/java/org/codice/ddf/security/crl/generator/CrlGenerator.java

```diff
@@ -0,0 +1,205 @@
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.security.crl.generator;
+
+import com.google.common.annotations.VisibleForTesting;
+import ddf.security.common.audit.SecurityLogger;
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Properties;
+import javax.ws.rs.core.Response;
+import org.apache.cxf.helpers.IOUtils;
+import org.apache.cxf.jaxrs.client.WebClient;
+import org.apache.wss4j.common.crypto.Merlin;
+import org.bouncycastle.util.Strings;
+import org.codice.ddf.configuration.AbsolutePathResolver;
+import org.codice.ddf.cxf.client.ClientFactoryFactory;
+import org.codice.ddf.cxf.client.SecureCxfClientFactory;
+import org.codice.ddf.platform.util.properties.PropertiesLoader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CrlGenerator implements Runnable {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(CrlGenerator.class);
+
+  static final String CRL_PROPERTY_KEY = Merlin.OLD_PREFIX + Merlin.X509_CRL_FILE;
+
+  static String issuerEncryptionPropertiesLocation =
+      new AbsolutePathResolver("etc/ws-security/issuer/encryption.properties").getPath();
+
+  static String issuerSignaturePropertiesLocation =
+      new AbsolutePathResolver("etc/ws-security/issuer/signature.properties").getPath();
+
+  static String serverEncryptionPropertiesLocation =
+      new AbsolutePathResolver("etc/ws-security/server/encryption.properties").getPath();
+
+  static String serverSignaturePropertiesLocation =
+      new AbsolutePathResolver("etc/ws-security/server/signature.properties").getPath();
+
+  static String crlFileLocation = new AbsolutePathResolver("etc/keystores").getPath();
+
+  static final String DEM_CRL = "/localCrl.crl";
+
+  static final String PEM_CRL = "/localCrl.pem";
+
+  private static final PropertiesLoader PROPERTIES_LOADER = PropertiesLoader.getInstance();
+
+  private static final String HTTPS = "https://";
+
+  private String crlLocationUrl;
+
+  private boolean crlByUrlEnabled;
+
+  private ClientFactoryFactory factory;
+
+  public CrlGenerator(ClientFactoryFactory factory) {
+    this.factory = factory;
+  }
+
+  /** Pulls down the CRL file provided via URL and writes it locally. */
+  @Override
+  public void run() {
+    if (crlByUrlEnabled && crlLocationUrl != null && crlLocationUrl.startsWith(HTTPS)) {
+      try {
+        SecureCxfClientFactory cxfClientFactory =
+            factory.getSecureCxfClientFactory(crlLocationUrl, WebClient.class);
+        WebClient client = cxfClientFactory.getWebClient();
+        Response response = client.get();
+
+        try (ByteArrayInputStream inputStream =
+            IOUtils.loadIntoBAIS((InputStream) response.getEntity())) {
+          File crlFile;
+
+          byte[] buffer = new byte[inputStream.available()];
```

Comment:
❓ Why all of this since you already loaded everything in a byte array input stream. You therefore have accessed to the whole `byte[]`